libs/store/src/lib/infrastructure/persistence/store/entity-collection-service-base.ts

```diff
@@ -0,0 +1,250 @@
+                                // entitySubject.error(error)
+                            },
+                            complete: () => entitySubject.complete()
+                        });
+
+                    // Listen to complete event in order to unsubscribe from "merge" stream above
+                    entitySubject.subscribe({
+                        complete: () => {
+                            subRequestCompletedSubject.next();
+                            subRequestCompletedSubject.complete();
+                        }
+                    });
+
+                    // Return subject so we are able to inject required data later
+                    return entitySubject;
+                })
+            );
+        }
+
+        // Proxy result subject.
+        const resultSubject$ = new ReplaySubject<T>(1);
+
+        // In order to run Request Chaining pipe it's needed to subscribe to it
+        // and push values in proxy subject.
+        // Once all sub entities are loaded this subscription will be completed automatically
+        entitySource.subscribe(resultSubject$);
+
+        return resultSubject$.asObservable();
+    }
+
+    /** @hidden */
+    private getSubEntityPrimaryKey<K extends BaseEntity>(
+        { key: keyOrFunction }: ChainingPolicyFieldOptions<T, K>,
+        entity: T
+    ): IdentityKey {
+        if (typeof keyOrFunction === 'function') {
+            return keyOrFunction(entity);
+        }
+        return (entity as T & { [key: string]: IdentityKey })[keyOrFunction];
+    }
+
+    /**
+     * Make request to Sub Entity and get back alive stream of this entity
+     *
+     * @param chainingEntries [keyToSubEntity: string, SubEntityChainingPolicy]
+     * @param entity Primary entity instance
+     * @returns sub entity alive stream
+     */
+    private createChainingRequests(
+        chainingEntries: [string, ChainingPolicyFieldOptions<T, any>][],
+        entity: T
+    ): Observable<{
+        [x: string]: any;
+    }>[] {
+        const requests = chainingEntries.map(([key, chainingOptions]) => {
+            const SubEntityClass = Array.isArray(chainingOptions.type) ? chainingOptions.type[0] : chainingOptions.type;
+
+            const subEntityService = this.entityCollectionsService.getEntityCollectionService(SubEntityClass);
+
+            // One entity requires using getByKey()
+            if (!Array.isArray(chainingOptions.type)) {
+                // get primary key
+                const subEntityKey = this.getSubEntityPrimaryKey(chainingOptions, entity);
+                return subEntityService.getByKey(subEntityKey).pipe(
+                    map((subEntity) => ({
+                        [key]: subEntity
+                    }))
+                );
+            }
+            // If it's array we have to request collection of entities
+            // TODO: how distinguish which one to use: getAll() or getWithQuery()?
+            // For now I use getAll() for simplicity
```

Comment:
We need to check here some usecases how it would behave. e..g 

class User {favColors: string[]}. might be ok, as you already get all, but 

class User {taskLists: Task[]} might be a problem, but for now unless we understand different usage scenarios should be good.